File: 2044. Count Number of Maximum Bitwise-OR Subsets.cpp
class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) 
    {
        unordered_map<int, int> count;
        int n = nums.size(), size = (1 << n), maxOr = 0;;
        for(int i = 0; i < size; i++)
        {
            int Or = 0;
            for(int j = 0; j < n; j++)
                if(i & (1 << j)) Or |= nums[j];
                
            count[Or] += 1;
            maxOr = max(maxOr, Or);
        }
        return count[maxOr];
    }
};

File: 2044. Count Number of Maximum Bitwise-OR Subsets.py
class Solution:
    def getOr(self, arr, Or=0):
        for num in arr:
            Or |= num
        return Or
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        n = len(nums)
        maxOr = self.getOr(nums)
        def backtrack(i, cur):
            if i == n:
                return self.getOr(cur) == maxOr
            not_take = backtrack(i + 1, cur)
            take = backtrack(i + 1, cur + [nums[i]])
            return take + not_take
        return backtrack(0, [])